### PR TITLE
Decode Avid private string parameters (SubCap captions, fonts, etc)

### PR DESCRIPTION
**Pull request by hasielhassan**
**Link the Issue(s) this Pull Request is related to.**

Not directly, just a need while trying to read SubCap captions out of an Avid export.

But I also know this other issue from some years ago that might be getting traction again?
https://github.com/AcademySoftwareFoundation/OpenTimelineIO/issues/62

**Summarize your change.**

Avid stores the string parameters of its plugins in a private byte array rather than an AAF string. pyaaf2 returns those as a list of ints, which `_transcribe_property` turns into an empty dict, so the text is lost:

```
"Caption": {}    ->    "Caption": "sq010_sht010-Á"
"Font": {}       ->    "Font": "Arial"
```

Changes:
- Added `_decode_avid_string_blob`, which parses that byte array and returns `None` if anything doesn't look right, so unrecognized values are left exactly as they are today
- Called it from `_transcribe_property` rather than a SubCap-specific helper, since the encoding seems to depend on the parameter type rather than the effect — this means Timecode Burn-In gets picked up too, and technically others too, like AvidTitler+, but currently explicitly excluding those since might prefer an extended custom parsing later.
- Skipped parameters that use the same container but hold non-text payloads (Timecode Burn-In `Effect State`, Avid Titler+ `MQP Blob`)
- Added `tests/sample_data/avid_string_blobs.aaf`, exported from Media Composer with SubCap, Timecode Burn-In and Avid Titler+, including accented and CJK characters on the captions

I also manually test all the test aaf files on the repo and noticed new updated parameters that are now stings instead of empty dicts

I initially had the code parsing using regex by poking at files rather than from any spec, then rebuilt the code with the help of AI to actually parse the binary data, its my first time with that, so I may well have gotten something wrong.

I'd really appreciate any feedfack, especially on the list of GUIDs I treat as text. Happy to change anything.

**Reference associated tests.**
Two test methods in the existing test script, checking the data in `tests/sample_data/avid_string_blobs.aaf`
- test_subcap_captions
- test_timecode_burn_in_text_parameters

### CODE REVIEW

**Comment by timlehr** [comment]
Hey @hasielhassan, thank you for the work, that's pretty cool.
Have you explored what it would take to properly round trip these values? Since we dont have a standard for this type of data yet, it would be awesome if we could round trip it successfully from the Avid into OTIO and back into the Avid.

File: src/otio_aaf_adapter/adapters/advanced_authoring_format.py [hunk 1]
```diff
@@ -90,6 +92,103 @@ def _get_class_name(item):
...
+    # magic(4s), payload_size(I), bom(2s), version(I), bom2(2s), guid(16s), chunk_size(I), str_len(I)
+    HEADER_FORMAT = "4sI2sI2s16sII"
+    HEADER_SIZE = 40
+
+    # Type GUIDs observed carrying a single NUL-terminated UTF-8 string. 
+    # Other GUIDs (notably c02f2caa-7293-11d6-b4e4-0030658a6504) carry 
+    # opaque, "plugin-defined" payloads, a UTF-16 struct, a table of GUIDs, 
+    # or an embedded Marquee document. 
+    # We currently only decode the following as one string:
+    AVID_STRING_TYPE_GUIDS = frozenset([
```
Comment (timlehr) [inline src/otio_aaf_adapter/adapters/advanced_authoring_format.py]: You should be able to store the UUIDs as a uuid.UUID object here to prevent the casting.

File: tests/test_aaf_adapter.py [hunk 2]
```diff
@@ -1851,6 +1855,42 @@ def test_non_av_track_kind(self):
                          ["Video", "AAF_DataEssenceTrack"]
                          )
 
+    def test_subcap_captions(self):
```
Comment (timlehr) [inline tests/test_aaf_adapter.py]: I am a bit concerned that the test might not cover enough different types of captions yet and could break depending on the AAF. Did you run the code against non-synthetic a bunch of production data too?

**Comment by hasielhassan** [comment]
Hey @timlehr thanks for the feedback.

Sorry for the late reply, I switched jobs and the last couple weeks were fun.

I understand the idea of the full roundtrip, I can give it a try in a couple weeks I think, I did explore the idea already and know is technically possible in terms of the reverse encoding, but I'm not sure of Avid will accept it easily so there is where I will need to spend most of the time.

I currently not have access to Media Composer, but can try to setup a VM when I get time and can give a try with the trial to see if I can make work.

Will keep you posted!
